route/build/src/histVars_data.f90

```diff
@@ -260,4 +275,184 @@ SUBROUTINE clean(this)
+                      ioDescHruHist)
+
+      ! reading basin runoff (hru dimension)
+      if (meta_hflx(ixHFLX%basRunoff)%varFile) then
+        allocate(this%basRunoff(this%nHru), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%basRunoff]'; return; endif
+
+        call read_pnetcdf(pioFileDesc, meta_hflx(ixHFLX%basRunoff)%varName, this%basRunoff, ioDescHruHist, ierr, cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+      end if
+
+      ! reading instantaneous runoff (rch dimension)
+      if (meta_rflx(ixRFLX%instRunoff)%varFile) then
+        allocate(this%instRunoff(this%nRch), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%instRunoff]'; return; endif
+
+        call read_pnetcdf(pioFileDesc, meta_rflx(ixRFLX%instRunoff)%varName, array_tmp, ioDescRchHist, ierr, cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+        ! need to shift tributary part in main core to after halo reaches (nTribOutlet)
+        if (masterproc) then
+          this%instRunoff(1:nRch_mainstem) = array_tmp(1:nRch_mainstem)
+          this%instRunoff(nRch_mainstem+nTribOutlet+1:this%nRch) = array_tmp(nRch_mainstem+1:nRch_mainstem+nRch_trib)
+        else
+          this%instRunoff = array_tmp
+        end if
+      end if
+
+      ! reading overland routed runoff (rch dimension)
+      if (meta_rflx(ixRFLX%dlayRunoff)%varFile) then
+        allocate(this%dlayRunoff(this%nRch), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%dlayRunoff]'; return; endif
+
+        call read_pnetcdf(pioFileDesc, meta_rflx(ixRFLX%dlayRunoff)%varName, array_tmp, ioDescRchHist, ierr, cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage); return; endif
+
+        ! need to shift tributary part in main core to after halo reaches (nTribOutlet)
+        if (masterproc) then
+          this%dlayRunoff(1:nRch_mainstem) = array_tmp(1:nRch_mainstem)
+          this%dlayRunoff(nRch_mainstem+nTribOutlet+1:this%nRch) = array_tmp(nRch_mainstem+1:nRch_mainstem+nRch_trib)
+        else
+          this%dlayRunoff = array_tmp
+        end if
+      end if
+
+      ! reading reach routed runoff and volume
+      if (nRoutes>0) then
+        allocate(this%discharge(this%nRch, nRoutes), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%discharge]'; return; endif
+        allocate(this%volume(this%nRch, nRoutes), stat=ierr, errmsg=cmessage)
+        if(ierr/=0)then; message=trim(message)//trim(cmessage)//' [hVars%discharge]'; return; endif
+
+        this%discharge = 0._sp
+        this%volume    = 0._sp
+
+        do ixRoute=1,nRoutes
+          select case(routeMethods(ixRoute))
+            case(accumRunoff);           ixFlow=ixRFLX%sumUpstreamRunoff
```

Comment:
Personally I think it's more clear to put on seperate lines rather than using ";"...